### PR TITLE
Use Documenter 0.20 for docs

### PR DESCRIPTION
**Pull request by iamed2**
Uses the new format, adds linkcheck, doctests, and a separate Travis stage for building docs.

### CODE REVIEW

File: docs/src/arithmetic.md [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
+
 ## ZonedDateTime-Period Arithmetic
 
-`ZonedDateTime` uses calendrical arithmetic in a [similar manner to `DateTime`](https://docs.julialang.org/en/stable/manual/dates/#TimeType-Period-Arithmetic-1) but with some key differences. Lets look at these differences by adding a day to March 30th 2014 in Europe/Warsaw.
+`ZonedDateTime` uses calendrical arithmetic in a [similar manner to `DateTime`](https://docs.julialang.org/en/stable/stdlib/Dates/#TimeType-Period-Arithmetic-1) but with some key differences. Lets look at these differences by adding a day to March 30th 2014 in Europe/Warsaw.
 
-```julia
-julia> using Base.Dates
+```jldoctest spring
+julia> using TimeZones, Compat.Dates
```
Comment (omus) [inline docs/src/arithmetic.md]: Can we just have the doctests run against Julia 1.0?

File: docs/src/conversions.md [hunk 2]
```diff
@@ -1,8 +1,17 @@
+# Conversions
+
+```@meta
+DocTestSetup = quote
+	using Compat.Dates
+	using TimeZones
```
Comment (omus) [inline docs/src/conversions.md]: Use spaces for indentation?

File: docs/src/arithmetic.md (hunk 1 above)
Comment (omus) [reply docs/src/arithmetic.md]: I think I'd prefer: `using TimeZones, Dates`

File: docs/src/current.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+You should be careful not to use `today()` when working with `ZonedDateTime`s as you may end up using the wrong day. For example:
+
+```@repl tz
+midway, apia = tz"Pacific/Midway", tz"Pacific/Apia"
+ZonedDateTime(today() + Time(11), midway)
+ZonedDateTime(today() + Time(11), apia)  # Should be 2018-01-30
+ZonedDateTime(today(midway) + Time(11), midway)
+ZonedDateTime(today(apia) + Time(11), apia)
+```
```
Comment (omus) [inline docs/src/current.md]: The output was important here. We need to fake the output here so the comment makes sense.

File: docs/src/arithmetic.md (hunk 1 above)
Comment (iamed2) [reply docs/src/arithmetic.md]: Sure

File: docs/src/current.md (hunk 3 above)
Comment (iamed2) [reply docs/src/current.md]: The output is generated for the built product. 

screenshot 2018-10-30 16 54 08

Comment (iamed2) [reply docs/src/current.md]: A `@repl` block generates the output in repl format when the docs are built

Comment (omus) [reply docs/src/current.md]: Can we dynamically update that comment?

Comment (omus) [reply docs/src/current.md]: Originally was:
```
julia> ZonedDateTime(today() + Time(11), apia)  # Should be 2018-01-30
2018-01-29T11:00:00+14:00
```

Comment (iamed2) [reply docs/src/current.md]: Oh I see, no. I can overwrite `today` for this block so it gives a fixed date

Comment (omus) [reply docs/src/current.md]: Alternatively we can change the comment to: `# Wrong date. With the current rules apia should be one day ahead of midway`

Comment (iamed2) [reply docs/src/current.md]: I like that better; done

File: .travis.yml [hunk 4]
```diff
@@ -33,3 +33,12 @@ after_success:
...
       Coveralls.submit(Coveralls.process_folder())'
+jobs:
+  include:
+    - stage: "Documentation"
+      julia: 1.0
+      os: linux
+      script:
+        - julia --project=docs/ -e 'using Pkg; Pkg.instantiate(); Pkg.develop(PackageSpec(path=pwd())); Pkg.build("TimeZones")'
+        - julia --project=docs/ docs/make.jl
+      after_success: skip
```
[the remark this answers is not in the document]
Comment (iamed2) [reply .travis.yml]: Yeah we could just split the script in two and have two jobs. I'll do that.

**Pull request merged**